mlflow/gateway/client.py

```diff
@@ -200,11 +203,23 @@ def create_route(self, name: str, route_type: str, model: Dict[str, Any]) -> Rou
                 "running Gateway server.",
                 error_code=BAD_REQUEST,
             )
+        if model is None:
+            raise MlflowException(
+                "Missing required field: model",
+                error_code=BAD_REQUEST,
+            )
         payload = {
             "name": name,
-            "route_type": route_type,
             "model": model,
         }
+        if model["provider"].lower() != Provider.DATABRICKS:
+            if route_type is None:
+                raise MlflowException(
+                    "Missing required field: route_type",
+                    error_code=BAD_REQUEST,
+                )
+            else:
+                payload["route_type"] = route_type
```

Comment:
Do we need this validation? Can we let the gateway backend decide?

Reply:
+1 to this. It might be a bit onerous to keep track of validation changes for the Databricks-only API here if all of the other validation logic lives not-in-OSS (as well as version mismatches between the client and the backend changes could be challenging to keep in sync)

Reply:
Yup great call, thought early-failing in the client would help but yeah maintaining this will become cumbersome. Removed the validation